### PR TITLE
BACKLOG-12844 : add wip action and wip modal component

### PR DESCRIPTION
**Pull request by jsinovassin**
## JIRA

https://jira.jahia.org/browse/BACKLOG-12844

### CODE REVIEW

File: src/javascript/EditPanel/WorkInProgress/OpenWorkInProgressModal.action.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
+
+import React, {useContext} from 'react';
+import PropTypes from 'prop-types';
+import WorkInProgressDialog from './WorkInProgressDialog/WorkInProgressDialog';
+import {ComponentRendererContext} from '@jahia/ui-extender';
+
+export const OpenWorkInProgressModal = ({context, render: Render}) => {
```
Comment (FBerthelot) [inline src/javascript/EditPanel/WorkInProgress/OpenWorkInProgressModal.action.jsx]:
```suggestion
export const OpenWorkInProgressModal = ({context, render: Render, ...props}) => {
```

File: src/javascript/EditPanel/WorkInProgress/OpenWorkInProgressModal.action.jsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    const componentRenderer = useContext(ComponentRendererContext);
+
+    const {siteInfo} = context;
+    const closeDialog = () => {
+        componentRenderer.destroy('WorkInProgressDialog');
+    };
+
+    return (
+        <>
+            <Render context={{
```
Comment (FBerthelot) [inline src/javascript/EditPanel/WorkInProgress/OpenWorkInProgressModal.action.jsx]: Avoid passing everything in the context object. Use plain React props ;)

File: src/javascript/EditPanel/WorkInProgress/WorkInProgressDialog/WorkInProgressDialog.jsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    const handleApply = () => {
+        onApply();
+        // TODO BACKLOG-12845 Do it
+    };
+
+    const isApplyDisabled = () => {
+        return isWipContent === isWip;
+    };
+
+    // TODO add the link to the sub title
```
Comment (FBerthelot) [inline src/javascript/EditPanel/WorkInProgress/WorkInProgressDialog/WorkInProgressDialog.jsx]: no ticket related ?

File: src/javascript/EditPanel/WorkInProgress/WorkInProgressDialog/WorkInProgressDialog.jsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
+import React, {useState} from 'react';
+import {
+    Dialog,
+    DialogActions,
+    DialogTitle,
+    DialogContent
+} from '@material-ui/core';
+import {Button, Typography} from '@jahia/moonstone';
+import {Apps, Check} from '@material-ui/icons';
```
Comment (FBerthelot) [inline src/javascript/EditPanel/WorkInProgress/WorkInProgressDialog/WorkInProgressDialog.jsx]: The icon are not in moonstone project ?

File: src/javascript/EditPanel/WorkInProgress/OpenWorkInProgressModal.action.jsx (hunk 2 above)
Comment (dgriffon) [reply src/javascript/EditPanel/WorkInProgress/OpenWorkInProgressModal.action.jsx]: I don't  agree on that one. the context contain all props that will be used to render the Render component properly, if we don't pass them all, we might miss some (as an action my have no knowledge of its usage)

**Pull request merged**